### DISCUSSION TITLE
Application

### DISCUSSION THREAD

**Discussion by Peio47** [robot-soccer-kit/robot-soccer-kit#43] [Q&A] [answered]
Bonjour,
Tout d'abord merci encore pour le travail fait.
J'ai testé l'application sur mon PC et elle fonctionne bien.
Par contre je l'ai testé dans mon lycée. Le game controller démarre mais les appli cliente ne donne rien. Elles se lancent, et continue sans s'arrêter ou émettre une erreur.
Est-ce que cela viendrait de la version de Python (3.7.4) ? Est-ce que cela pourrait venir des restriction ?

En vous remerciant
Pierre

**Comment by cgaspard3333** [comment]
Bonjour,
Pouvez-vous tout d'abord me préciser quel est le système d'exploitation des postes concernés (Game Controller et Clients).
De quelle manière sont ils raccordés ensemble au même réseau ? 
La version de Python 3.7.4 pourrait aussi potentiellement être un problème sachant que les tests de compatibilités menés donnent Python 3.8.X ou 3.9.X afin d'obtenir un fonctionnement optimal.
Cependant, je me tourne plutôt vers un problème de restrictions réseau inhérente à Windows (si c'est bien cet OS dont vous disposez) ou au réseau lui même.
Je vais lancer des tests sur des postes Windows 10 en Game Controller et Clients et en attendant pour voir si j'arrive à reproduire le problème et à en trouver la source. 

Je reprécise au cas où qu'il faut bien penser à changer l'adresse IP (dans l'exemple 127.0.0.1 lorsqu'on contrôle les robots depuis le même PC où tourne le Game Controller) pour la faire correspondre à celle du PC "serveur" faisant tourner le Game Controller : 
```
import rsk

with rsk.Client(host='127.0.0.1', key='') as client:
    client.robots['green'][1].kick()
```

**Comment by Peio47** [comment]
Les deux applications fonctionnent avec deux ordi sous Windows (10 me semble-t-il), reliés par switch.
Le problème survient également quand les deux appli tournent sur le même ordi, et ce avec un compte admin (mais avec quand même un certain nombre de restrictions)

**Reply by cgaspard3333** [reply]
Le fait qu'on ne puisse pas le faire sur le même PC me parait bizarre.
Par contre effectivement, en utilisant un Game Controller Windows, il semblerait que quelque chose bloque lorsque les clients essaient de se connecter au Game Controller lors de mes tests. 
Je vous tiens informé quant à la résolution de ce problème. 

Pas de problèmes avec un Game Controller Linux et des Clients Windows ou Linux d'après mes tests.

**Comment by cgaspard3333** [comment, marked as answer]
Je viens de terminer mes tests et de trouver le problème.

1. La vision doit être fonctionnelle afin de pouvoir contrôler les robots à l'aide d'un client. Si vous souhaitez tester de contrôler les robots sans que la vision soit opérationnelle il faut écrire : 
`with rsk.Client(host='127.0.0.1', key='', wait_ready=False) as client:` 
au lieu de 
`with rsk.Client(host='127.0.0.1', key='') as client:`
En effet, rajouter l'argument `wait_ready=False` permet de ne pas avoir à lancer la vision du Game Controller pour pouvoir envoyer une commande aux robots depuis les Clients (Attention car un grand nombre de commandes dépendent de la vision)

2. Afin que les Clients puissent envoyer des messages au Game Controller, il est nécessaire que : 
    a. Soit l'application Python soit débloquée dans le pare-feu Windows dans le trafic entrant et sortant du PC Game Controller.
    b. Soit que les ports 7557 et 7558 soient débloqués dans le pare-feu Windows dans le trafic entrant et sortant du PC Game Controller.

**Reply by Peio47** [reply]
Bonjour,
Grâce à tes indications j'ai pu faire communiquer le client et le serveur quand s'ils sont sur un même PC.
En revanche pas moyen de les faire communiquer sur deux postes séparés, les pare-feux Windows étant désactivés. 
On va essayer avec des mini PC sous Linux avec un réseau séparé de celui du lycée.

**Reply by cgaspard3333** [reply]
Bonjour, 
Nous sommes entrain de travailler pour faire en sorte que les personnes s'occupant des services informatiques des lycées de Nouvelle-Aquitaine débloquent les ports que nous utilisons le plus rapidement possible dans les lycées concernés. Cependant, comme vous le savez sans doute, l'inertie peut être un peu longue du côté de la région.
Je reviens vers vous dès que j'ai des nouvelles à ce propos. 
Tenez moi au courant si vous y arrivez avec des Mini PC Linux. Au passage, normalement, même de simples Raspberry Pi suffisent côté client, la quasi totalité des calculs se réalisant côté serveur.
